Client/Frontend/Browser/Tab.swift

```diff
@@ -492,6 +492,19 @@ class Tab: NSObject {
         checkTabCount(failures: 0)
         #endif
     }
+    
+    /// When a user clears ALL history, `sessionData` and `historyList` need to be purged, and close the webView.
+    func clearAndResetTabHistory() {
+        guard let currentlyOpenUrl = sessionData?.urls.last ?? historyList.last ?? url else { return }
+        
+        url = currentlyOpenUrl
+        sessionData = SessionData(currentPage: 0, urls: [currentlyOpenUrl], lastUsedTime: Date.now())
+        sessionData?.urls = migrate(urls: [currentlyOpenUrl])
```

Comment:
Q. what made you migrate urls that too currently opened one?
I am guessing it has internal, if so can we add an explanation as to why we need to migrate

Reply:
Double checked - this works without migration!